docs/src/utils.md

```diff
@@ -23,3 +23,26 @@ Zygote.checkpointed
 ```
 
 `Params` and `Grads` can be copied to and from arrays using the `copy!` function.
+
+### Operations with Grads
+
+Map and broadcast are supported for the dictionary-like `Grads` object.
+```julia
+using Zygote, Test
+
+w = rand(2)
+x1 = rand(2)
+x2 = rand(2)
+b = rand(2)
+
+gs1 = gradient(() -> sum(w .* x1 .+ b), Params([w])) 
+gs2 = gradient(() -> sum(w .* x2  .+ b), Params([w])) 
+
+# accumulate gradients
+gs = gs .+ gs
+@test gs[w] ≈ gs1[w] + gs2[w] 
+@test gs[b] ≈ gs1[b] + gs2[b] 
+
+# clip gradients in-place
+map!(x -> clamp!(x, -0.1, 0.1), gs, gs)
```

Comment:
Ideally this would be `foreach(x -> clamp!(x, -0.1, 0.1), gs)`, maybe that should be allowed?

Reply:
we should define iteration at this point. 
Julia's `Dict`s iterate over pairs. 
While dictionaries here https://github.com/andyferris/Dictionaries.jl iterate over values. 
Seems more convenient for us to iterate over values, 
and use `pairs(gs)` if one wants pairs.

Reply:
Following base conventions is important to keep the same semantics for when you have arbitrary code samples to contend with. Yeah, Julia should potentially have a method to iterate over values, but seems weird to have a dictionary when you need the values and not the keys.

Reply:
You can already iterate `Grads` with `pairs(gs)`. Given that we are using `IdDict`, it makes sense to stick with those semantics.

`Params` has an `order` field, does that get respected by `pairs(gs)` (I don't think so).

Reply:
The reason for iterating over pairs is that ordering is not guaranteed. Those iteration semantics prevent accidentally iterating two dictionaries thinking their ordering is the same.

I think the same applies here. If two `Grads` have the same `Params` in different orders, should we be able to add them? I think so.

Reply:
`Grads` doesn't subtype anything, so it doesn't seem too crazy for it to support a limited range of operations we think useful. It's just a container to hold particular results.

If it was made iterable, writing `for (p,g) in grads; ...` might sometimes be tidier than `for p in params; grads[p] ...`. It does expose a bit more, in that you're explicitly using the fact that it's really a grads-and-params struct. 

Its `Params` does provide an order. Although as you say, the same parameters wrapped in a different `Params` struct could have a different order. So `grads1 .+ 0.1 .* grads2` would probably still want to disobey this order and go by the keys.

Reply:
You should already be able to do `for (p, g) in grads` (I was doing this yesterday when playing around with alternate implementations of this PR).

Reply:
Ah never mind, I see from the latest commits that maybe this includes extraneous pairs somehow?

Reply:
yes, this is what I remember from inspecting Grads objects in the past, altough i'm not 100% sure

Reply:
This is better anyways since I am not sure the ordering of `grads.params` was being respected before.

Reply:
I didn't think it was iterable at all:
```
julia> g0 = Zygote.Grads(IdDict(), Zygote.Params())
Grads(...)

julia> (g0...,)
ERROR: MethodError: no method matching iterate(::Zygote.Grads)
```

Reply:
Woah you're right. I must have been mistakenly operating on the `IdDict` I created? I was surprised to see it worked and assumed it happened by some Base magic since `haskey` was forwarded. Guess I was wrong!

Reply:
I think the value based iteration is the most natural here, since it's consistent with how I implemented map and is battle tested in Dictionaries.jl. I'll mark this conversion as resolved since it's on an outdated commit, but we can continue the discussion in the main thread